### PR TITLE
Add Reset Button in Share Modal in Entries Viz

### PR DESCRIPTION
**Pull request by subinasr**
Depends on: https://github.com/the-deep/server/pull/721

### CODE REVIEW

File: src/views/Entries/ShareModal/index.tsx [hunk 1]
```diff
@@ -72,6 +78,32 @@ function ShareModal(props: ComponentProps) {
...
+        resetResponse,
+        ,
+        triggerResetPublicUrl,
+    ] = useRequest<PublicUrl>(
+        {
+            url: `server://projects/${projectId}/public-viz/`,
+            method: 'POST',
+            body: resetBodyToSend,
+            onSuccess: (response) => {
+                console.warn('url resetted to', response.publicUrl);
```
Comment (AdityaKhatri) [inline src/views/Entries/ShareModal/index.tsx]: You should set this as the new URL.

File: src/views/Entries/ShareModal/index.tsx [hunk 2]
```diff
@@ -72,6 +78,32 @@ function ShareModal(props: ComponentProps) {
...
+        },
+    );
+
+    const handleResetClick = useCallback(() => {
+        setResetClicked(true);
+        triggerResetPublicUrl();
+    }, [triggerResetPublicUrl]);
+
+    console.warn('public Url publicUrlavailability', publicUrl, publicUrlAvailable);
+    console.warn('reset clicked', resetClicked);
```
Comment (AdityaKhatri) [inline src/views/Entries/ShareModal/index.tsx]: @subinasr ma'am why?

File: src/views/Entries/ShareModal/index.tsx [hunk 3]
```diff
@@ -46,11 +46,17 @@ function ShareModal(props: ComponentProps) {
     } = props;
 
     const [publicUrlAvailable, setPublicUrlAvailability] = useState(isDefined(publicUrl));
+    const [resetClicked, setResetClicked] = useState<boolean>(false);
```
Comment (AdityaKhatri) [inline src/views/Entries/ShareModal/index.tsx]: Why is this necessary?

File: src/views/Entries/ShareModal/index.tsx [hunk 4]
```diff
@@ -91,17 +123,29 @@ function ShareModal(props: ComponentProps) {
...
+                                label={_ts('entries', 'getPublicLinkLabel')}
+                                disabled={pending}
+                            />
+                            { publicUrlAvailable && (
+                                <Button
+                                    className={styles.reset}
+                                    iconName="circle"
+                                    onClick={handleResetClick}
+                                >
+                                    Reset
```
Comment (AdityaKhatri) [inline src/views/Entries/ShareModal/index.tsx]: Use strings

File: src/views/Entries/ShareModal/index.tsx [hunk 5]
```diff
@@ -72,6 +74,43 @@ function ShareModal(props: ComponentProps) {
...
+            e?.clipboardData?.setData('text/plain', publicUrl ?? '');
+            e.preventDefault();
+        });
+        document.execCommand('copy');
+
+        notify.send({
+            title: _ts('entries', 'copiedToClipboard'),
+            type: notify.type.SUCCESS,
+            message: _ts('entries', 'copiedToClipboardSuccessfully'),
+            duration: notify.duration.MEDIUM,
```
Comment (AdityaKhatri) [inline src/views/Entries/ShareModal/index.tsx]: Lets change this to FAST

File: src/views/Entries/ShareModal/index.tsx [hunk 6]
```diff
@@ -72,6 +74,43 @@ function ShareModal(props: ComponentProps) {
...
+                }
+            },
+            onFailure: (_, errorBody) =>
+                notifyOnFailure(_ts('entries', 'resetVizLink'))({ error: errorBody }),
+        },
+    );
+
+    const pending = pendingUrlVisibilityChange || pendingReset;
+
+    const copyToClipboard = () => {
```
Comment (tnagorra) [inline src/views/Entries/ShareModal/index.tsx]: Please use useCallback.

File: src/views/Entries/ShareModal/index.tsx [hunk 7]
```diff
@@ -72,6 +74,43 @@ function ShareModal(props: ComponentProps) {
...
+        document.addEventListener('copy', (e) => {
+            e?.clipboardData?.setData('text/plain', publicUrl ?? '');
+            e.preventDefault();
+        });
```
Comment (tnagorra) [inline src/views/Entries/ShareModal/index.tsx]: Do we not have simpler method to copy to clipboard.
Also, we need to remove event listener

**Review by tnagorra — CHANGES_REQUESTED** [review]
Copy mechanism needs more work!
